### PR TITLE
Fix: Find in Files results don't update to reflect external changes unless file is open

### PR DESCRIPTION
**Pull request by amrelnaggar**
This addresses #10717 and covers all FileSystem change events

### CODE REVIEW

**Comment by tallandroid** [comment]
Let me know if its accepted. I will stop working on it if thats the case.

**Comment by zaggino** [comment]
LGTM

File: src/search/FindInFiles.js [hunk 1]
```diff
@@ -795,7 +795,7 @@ define(function (require, exports, module) {
         
         var addPromise;
         if (entry.isDirectory) {
-            if (!added || !removed) {
+            if (!added || !removed || (!added.length && !removed.length)) {
```
Comment (zaggino) [inline src/search/FindInFiles.js]: I'd personally prefer ` || (added.length === 0 && removed.length === 0))`

**Comment by amrelnaggar** [comment]
Made the changes, this was better indeed.

**Comment by zaggino** [comment]
Thanks @amrelnaggar !

**Pull request merged**